### PR TITLE
First attempt to make download work with new DSL2 syntax.

### LINKED ISSUE #1379: nf-core download doesn't work with latest NF syntax
### Description of the bug

```console
$ nf-core download    

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.2



Specify the name of a nf-core pipeline or a GitHub repository name (user/repo).
? Pipeline name: rnaseq
? Select release / branch: 3.5  [release]

In addition to the pipeline code, this tool can download software containers.
? Download software container images: singularity

If transferring the downloaded files to another system, it can be convenient to have everything compressed in a single file.
This is not recommended when downloading Singularity images, as it can take a long time and saves very little space.
? Choose compression type: none
INFO     Saving 'nf-core/rnaseq'                                                                                                                                       download.py:160
          Pipeline revision: '3.5'                                                                                                                                                    
          Pull containers: 'singularity'                                                                                                                                              
          Using $NXF_SINGULARITY_CACHEDIR': /home/harshil/containers/singularity                                                                                                      
          Output directory: 'nf-core-rnaseq-3.5'                                                                                                                                      
INFO     Downloading workflow files from GitHub                                                                                                                        download.py:163
INFO     Downloading centralised configs from GitHub                                                                                                                   download.py:167
INFO     Found 1 container                                                                                                                                             download.py:472
Pulling singularity images ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% • 0/1 completed
${ workflow.containerEngine ==  Copying from cache to target directory ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
CRITICAL [Errno 2] No such file or directory: '/home/harshil/containers/singularity/${ workflow.containerEngine == .img'                                               download.py:182
(nf-core) ~/testing 
```

The regex in the [nf-core download](https://github.com/nf-core/tools/blob/d0ab20e5f31a40ed81c26a5eab836093b9297115/nf_core/download.py) script will need to be updated (and to be backward compatible).

### Command used and terminal output

Try running `nf-core download` to fetch Singularity containers for v3.5 of nf-core/rnaseq.

### PR DESCRIPTION
**Pull request by ewels**
Update the code-scraping-code that discovers container URLs in DSL2 module code. Hopefully now works with both old and new DSL2 styles. Should close #1379.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by ewels** [comment]
Opening this a little early as I'm still technically ill and not working, but want to get this moving ASAP.
⚠️  Currently completely untested.

Any help with testing much appreciated. Apologies in advance for dumb errors. I'm going to go and take a nap now. 🤒 😴

**Comment by ewels** [comment]
Ok I lied, I didn't go to bed yet and did some very quick testing to make sure that it runs (it didn't).

Speed test looks good at my end now:

✅  Finds 28 containers for nf-core/rnaseq version 3.2 (old DSL2 syntax)
✅  Finds 30 containers for nf-core/rnaseq version 3.5 (new DSL2 syntax)
✅  Finds 1 non-https container for nf-core/methylseq (DSL1)

More testing by others would be much appreciated 👍🏻

File: nf_core/download.py [hunk 1]
```diff
@@ -439,35 +459,36 @@ def find_container_images(self):
...
-                        else:
-                            if len(matches) > 0:
-                                containers_raw.append(matches[0].strip('"').strip("'"))
+                        this_container = None
+                        contents = fh.read()
+                        matches = re.findall(r"container\s*\"([^\"]*)\"", contents, re.S)
+                        if matches:
+                            for match in matches:
+                                # Look for a http download URL.
+                                # Thanks Stack Overflow for the regex: https://stackoverflow.com/a/3809435/713980
```
Comment (drpatelh) [inline nf_core/download.py:468]: 🙇🏽

File: nf_core/download.py [hunk 2]
```diff
@@ -439,35 +459,36 @@ def find_container_images(self):
...
+                                url_regex = r"https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)"
```
Comment (drpatelh) [inline nf_core/download.py:469]: 🤯

**Review by drpatelh — APPROVED** [review]
I haven't had time to test it I'm afraid but if you think it's good to go then let's get it in. I will test when I get a moment.

**Pull request merged**